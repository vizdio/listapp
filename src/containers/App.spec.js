import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import MyConnectedComponent from './App';

const initialState = [
    {"created": 1572381350000, "name": "Super Mario Bros", "selected": false, "value": "85"},
    {"created": 1575382350000, "name": "Legend of Zelda, The", "selected": false, "value": "103"},
    {"created": 1571383350000, "name": "Metroid", "selected": false, "value": "69"},
    {"created": 1570384350000, "name": "Techmo Bowl", "selected": false, "value": "22"},
    {"created": 1572385350000, "name": "Final Fantasy", "selected": false, "value": "77"}
  ];

  const mockStore = configureStore([]);
  
  describe('My Connected React-Redux Component', () => {
    let store;
    let component;
    beforeEach(() => {
      store = mockStore(initialState);
      component = renderer.create(
        <Provider store={store}>
          <MyConnectedComponent />
        </Provider>
      );
    });
    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
  });