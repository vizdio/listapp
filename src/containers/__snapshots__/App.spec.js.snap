// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component should render with given state from Redux store 1`] = `
<div>
  <header
    className="header"
  >
    <h1>
      MY LIST APP
    </h1>
    <h2>
      React/Redux
    </h2>
  </header>
  <main>
    <ul
      className="list-container"
    >
      <li
        className="row"
        onClick={[Function]}
      >
        <div
          className="selected"
        >
          <svg>
            unchecked.svg
          </svg>
        </div>
        <div
          className="name"
        >
          Super Mario Bros
        </div>
        <div
          className="value"
        >
          <span>
            $85.00
          </span>
        </div>
        <time
          className="date"
          dateTime={"2019-10-29T20:35:50.000Z"}
        >
          October 29, 2019 4:35pm
        </time>
        <div
          className="delete"
        >
          <svg
            onClick={[Function]}
          >
            delete.svg
          </svg>
        </div>
      </li>
      <li
        className="row"
        onClick={[Function]}
      >
        <div
          className="selected"
        >
          <svg>
            unchecked.svg
          </svg>
        </div>
        <div
          className="name"
        >
          Legend of Zelda, The
        </div>
        <div
          className="value"
        >
          <span>
            $103.00
          </span>
        </div>
        <time
          className="date"
          dateTime={"2019-12-03T14:12:30.000Z"}
        >
          December 03, 2019 9:12am
        </time>
        <div
          className="delete"
        >
          <svg
            onClick={[Function]}
          >
            delete.svg
          </svg>
        </div>
      </li>
      <li
        className="row"
        onClick={[Function]}
      >
        <div
          className="selected"
        >
          <svg>
            unchecked.svg
          </svg>
        </div>
        <div
          className="name"
        >
          Metroid
        </div>
        <div
          className="value"
        >
          <span>
            $69.00
          </span>
        </div>
        <time
          className="date"
          dateTime={"2019-10-18T07:22:30.000Z"}
        >
          October 18, 2019 3:22am
        </time>
        <div
          className="delete"
        >
          <svg
            onClick={[Function]}
          >
            delete.svg
          </svg>
        </div>
      </li>
      <li
        className="row"
        onClick={[Function]}
      >
        <div
          className="selected"
        >
          <svg>
            unchecked.svg
          </svg>
        </div>
        <div
          className="name"
        >
          Techmo Bowl
        </div>
        <div
          className="value"
        >
          <span>
            $22.00
          </span>
        </div>
        <time
          className="date"
          dateTime={"2019-10-06T17:52:30.000Z"}
        >
          October 06, 2019 1:52pm
        </time>
        <div
          className="delete"
        >
          <svg
            onClick={[Function]}
          >
            delete.svg
          </svg>
        </div>
      </li>
      <li
        className="row"
        onClick={[Function]}
      >
        <div
          className="selected"
        >
          <svg>
            unchecked.svg
          </svg>
        </div>
        <div
          className="name"
        >
          Final Fantasy
        </div>
        <div
          className="value"
        >
          <span>
            $77.00
          </span>
        </div>
        <time
          className="date"
          dateTime={"2019-10-29T21:42:30.000Z"}
        >
          October 29, 2019 5:42pm
        </time>
        <div
          className="delete"
        >
          <svg
            onClick={[Function]}
          >
            delete.svg
          </svg>
        </div>
      </li>
    </ul>
    <div
      className="sort-container"
    >
      <button
        className="sort-button"
        onClick={[Function]}
      >
        SORT BY NAME
      </button>
      <button
        className="sort-button"
        onClick={[Function]}
      >
        SORT BY VALUE
      </button>
      <button
        className="sort-button"
        onClick={[Function]}
      >
        SORT BY DATE
      </button>
    </div>
    <div
      className="add-container"
    >
      <div>
         Name: 
        <input
          type="text"
        />
         
      </div>
      <div>
         Value: 
        <input
          type="text"
        />
         
      </div>
      <button
        className="add-button"
        onClick={[Function]}
      >
        ADD
      </button>
    </div>
  </main>
  <footer>
    <p>
      By Jason Smith
    </p>
    <small>
      Created for FormStack Front-End Engineer Assignment
    </small>
  </footer>
</div>
`;
